Configuration/FlexForms/frontendsearch_plugin.xml

```diff
@@ -101,5 +101,151 @@
+                </TCEforms>
+                <type>array</type>
+                <el>
+                    <settings.searchResultsFilter.title>
+                        <TCEforms>
+                            <exclude>1</exclude>
+                            <label>LLL:EXT:dpf/Resources/Private/Language/locallang.xlf:tt_content.pi_flexform.searchResultsFilter.title</label>
+                            <config>
+                                <type>input</type>
+                                <eval>trim</eval>
+                            </config>
+                        </TCEforms>
+                    </settings.searchResultsFilter.title>
+                    <settings.searchResultsFilter.author>
+                        <TCEforms>
+                            <exclude>1</exclude>
+                            <label>LLL:EXT:dpf/Resources/Private/Language/locallang.xlf:tt_content.pi_flexform.searchResultsFilter.author</label>
+                            <config>
+                                <type>input</type>
+                                <eval>trim</eval>
+                            </config>
+                        </TCEforms>
+                    </settings.searchResultsFilter.author>
+                    <settings.searchResultsFilter.abstract>
+                        <TCEforms>
+                            <exclude>1</exclude>
+                            <label>LLL:EXT:dpf/Resources/Private/Language/locallang.xlf:tt_content.pi_flexform.searchResultsFilter.abstract</label>
+                            <config>
+                                <type>input</type>
+                                <eval>trim</eval>
+                            </config>
+                        </TCEforms>
+                    </settings.searchResultsFilter.abstract>
+                    <settings.searchResultsFilter.tag>
+                        <TCEforms>
+                            <exclude>1</exclude>
+                            <label>LLL:EXT:dpf/Resources/Private/Language/locallang.xlf:tt_content.pi_flexform.searchResultsFilter.tag</label>
+                            <config>
+                                <type>input</type>
+                                <eval>trim</eval>
+                            </config>
+                        </TCEforms>
+                    </settings.searchResultsFilter.tag>
+                    <settings.searchResultsFilter.doctype>
+                        <TCEforms>
+                            <exclude>1</exclude>
+                            <label>LLL:EXT:dpf/Resources/Private/Language/locallang.xlf:tt_content.pi_flexform.searchResultsFilter.doctype</label>
+                            <config>
+                                <type>select</type>
+                                <renderType>selectMultipleSideBySide</renderType>
+                                <foreign_table>tx_dpf_domain_model_documenttype</foreign_table>
```

Comment:
This shows unfortunately all documenttype of all tenant configurations.

A sorting by name would be great, too. This could be done by something like
`<foreign_table_where>AND [...] ORDER BY tx_dpf_domain_model_documenttype.name</foreign_table_where>`

The displayName could be used. But this may be (only? be) changed in TCA-Configuration of  tx_dpf_domain_model_documenttype.

Reply:
The following would do the job:

1. Flexform
                                    <foreign_table_where>AND tx_dpf_domain_model_documenttype.pid = ###PAGE_TSCONFIG_ID### ORDER BY tx_dpf_domain_model_documenttype.display_name</foreign_table_where>

2. Page TSconfig on the root of the Tenant Tree
     TCEFORM.tt_content.pi_flexform.PAGE_TSCONFIG_ID = XXX

3. To show displayName in Flexform and Listmodule change in ext_tables.php:
    GLOBALS['TCA']['tx_dpf_domain_model_documenttype'] = array(
    'ctrl' => array(
        'title'                    => 'LLL:EXT:dpf/Resources/Private/Language/locallang_db.xlf:tx_dpf_domain_model_documenttype',
        **'label'                    => 'display_name',**